### PR TITLE
Task 3 part 2

### PR DESCRIPTION
**Pull request by KrisKulbych**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Run checks
+        run: |
+          isort --check --diff .
```
Comment (DivnychAndrii) [inline .github/workflows/build.yaml]: Please separate one check per one - name step. For the sake of atomicy

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          pip install -r requirements.txt
+      - name: Run checks
+        run: |
+          isort --check --diff . 
+          black --check --diff .
+          flake8 
+          mypy .
+      - name: Run tests with Pytest
+        run: |
+          pytest
```
Comment (DivnychAndrii) [inline .github/workflows/build.yaml]: I would love to see the coverage as well. And the part of pytest check to fail the test run if coverage <90% https://pytest-cov.readthedocs.io/en/latest/config.html#reference

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+[tool.black]
+line_length = 79
```
Comment (DivnychAndrii) [inline pyproject.toml]: You can use 120

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[flake8]
```
Comment (DivnychAndrii) [inline setup.cfg]: Mypy black configs can go to a single pyproject. toml. Typically this is the file with all configs.  As far as I know flake8 does not support pyproject.toml, so let this file be here, but we will not use flake in the future tasks anyway

Comment (DivnychAndrii) [inline setup.cfg]: image
Requirements.txt
I don't see the usage of ﻿colorama==0.4.6, iniconfig==2.0.0 mccabe==0.7.0 packaging==24.1 pluggy==1.5.0 pycodestyle==2.12.1 pyflakes==3.2.0, please remove redundant packages from the requirements

File: tests/test_characters_counter.py [hunk 5]
```diff
@@ -5,21 +5,27 @@
...
-            self, input_text) -> None:
+        self, input_text: str
+    ) -> None:
         with pytest.raises(TypeError):
             count_characters_occurrence(input_text)
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
     pytest.main()
```
Comment (DivnychAndrii) [inline tests/test_characters_counter.py:22]: Please add github badge into readme

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/monitoring-workflows/adding-a-workflow-status-badge

File: pyproject.toml (hunk 3 above)
Comment (KrisKulbych) [reply pyproject.toml]: Ok, I decided to try and made changes to my config files.

File: setup.cfg (hunk 4 above)
Comment (KrisKulbych) [reply setup.cfg]: Ok, I removed by hand all connected packages from requirements.txt

File: .github/workflows/build.yaml (hunk 1 above)
Comment (KrisKulbych) [reply .github/workflows/build.yaml]: Done!

File: .github/workflows/build.yaml (hunk 2 above)
Comment (KrisKulbych) [reply .github/workflows/build.yaml]: Done!

File: setup.cfg (hunk 4 above)
Comment (KrisKulbych) [reply setup.cfg]: Allright, but I decided to replace configurations for flake8 from setup.cfg to .flake8. I'm just experimenting =)

File: tests/test_characters_counter.py (hunk 5 above)
Comment (KrisKulbych) [reply tests/test_characters_counter.py:22]: I added badge, thank you!

**Review by DivnychAndrii — APPROVED** [review]

**Pull request merged**